### PR TITLE
[WIP] Decorator prefix

### LINKED ISSUE #3: Collision with seneca
I'm currently using seneca and would like to try out hemera from the same hapi server (for a few routes) but I'm running into an issue with decorations as seneca is already defining them (via Chairo)

```shell
(node:9810) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 1): Error: Server decoration already defined: action
```

What do you think about extending `internals.defaults` with an option to disable hapi decorations. Obviously this would be disabled/false by default.

Then we can check the options before decorating hapi. I'll do a quick PR if you think this makes sense for the general hemera community.

### PR DESCRIPTION
**Pull request by sberryman**
Fixes #3

### CODE REVIEW

**Comment by sberryman** [comment]
@StarpTech - Prefix is implemented along with a few tests and updated readme. Let me know what needs to be changed to fit your style!

File: README.md [hunk 1]
```diff
@@ -44,15 +44,17 @@ server.register({
...
           expiresIn: 60000,
           staleIn: 30000,
           staleTimeout: 10000,
           generateTimeout: 100
         }
       }
-    }
+    },
+    // run chairo and hemera side-by-side
+    hapiDecoratePrefix: 'testPrefix'
```
Comment (StarpTech) [inline README.md]: Call it just `decoratePrefix`

File: README.md [hunk 2]
```diff
@@ -167,3 +169,41 @@ server.route({
 ```curl
  curl http://localhost:3000/foo/math/add?a=2&b=2
 ```
+
+prefix the hapi decorations (chairo and hemera co-exist)
```
Comment (StarpTech) [inline README.md]: Using chairo and hemera at the same is very special and I dont want to encourage them. Its good that a prefix exist but chairo or seneca should not be mentioned.

File: lib/index.js [hunk 3]
```diff
@@ -37,11 +38,9 @@ exports.register = function (server, options, next) {
     apply: true
   })
 
-  if (!settings.disableHapiDecorate) {
-    server.decorate('server', 'action', internals.action(server))
-    server.decorate('reply', 'act', internals.replyAct)
-    server.handler('act', internals.handlerAct)
-  }
+  server.decorate('server', [options.hapiDecoratePrefix || '', 'action'].join(''), internals.action(server))
```
Comment (StarpTech) [inline lib/index.js]: `${options.decoratePrefix}_${action}` much better

Comment (sberryman) [inline lib/index.js]: I agree, but what happens if no prefix is defined. then it turns into request._act or reply._act.

File: README.md (hunk 2 above)
Comment (sberryman) [inline README.md]: works for me, i'll pull out the readme changes

File: README.md (hunk 1 above)
Comment (sberryman) [inline README.md]: 👌🏼

File: lib/index.js [hunk 4]
```diff
@@ -38,9 +38,15 @@ exports.register = function (server, options, next) {
     apply: true
   })
 
-  server.decorate('server', [options.hapiDecoratePrefix || '', 'action'].join(''), internals.action(server))
-  server.decorate('reply', [options.hapiDecoratePrefix || '', 'act'].join(''), internals.replyAct)
-  server.handler([options.hapiDecoratePrefix || '', 'act'].join(''), internals.handlerAct)
+  if (options.decoratePrefix) {
+    server.decorate('server', `${options.decoratePrefix}_action`, internals.action(server))
+    server.decorate('reply', `${options.decoratePrefix}_act`, internals.replyAct)
+    server.handler(`${options.decoratePrefix}_act`, internals.handlerAct)
```
Comment (StarpTech) [inline lib/index.js]: I wouldn't prefix it with a `_` but ensure camelCase e.g

```
decoratePrefix = 'test'
testAction()
testAct()
testAdd()
```
the underscore was just an example. Thanks

**Comment by StarpTech** [comment]
Close PR due to inactivity and fixed with https://github.com/hemerajs/hapi-hemera/pull/6

**Pull request closed without merging**